src/state.py

```diff
@@ -11,6 +12,52 @@ def __init__(self):
     def add_student(self, student):
         self.students[student.student_id] = student
 
+    def get_score(self): 
+        score = 0
+        #if traded the subject as expected get score
+        for i in self.students.values(): 
+            #checking buddies 
+            for numbers in set(list(i.buddies.values())):
+                    for classes in set(list(i.buddies.keys())): 
+                        if i.subjects_and_classes[classes] == self.students[numbers].subjects_and_classes[classes]: 
+                            score += 30 
+                        else: 
+                            score -= 20
+                        
+
+            for position, j in enumerate(i.subjects_and_classes): 
+                if j in i.subject_targets.keys(): 
+                    # check if the student got the target class 
+                    if i.subjects_and_classes[j] in i.subject_targets[j]:  
+                        score += 40
+                    else: 
+                        score -= 30
+
+                if j in i.subject_give_ins.keys():
+                    # check the givin classes but not so much
+                    if i.subjects_and_classes[j] in i.subject_give_ins[j]: 
+                        score -= 3      #if a class was abdicated
+                    else: 
+                        score += 5
+                     
+
+               #checking schedule conflicts. If two classes are incompatible then score = -100000
+                for p in range(position+1, len(list(i.subjects_and_classes.keys()))):
+                    key = list(i.subjects_and_classes.keys())[p] 
+                    sched_1 = self.class_schedules[translate_subject_and_class(j,i.subjects_and_classes[j])]
+                    sched_2 = self.class_schedules[translate_subject_and_class(key, i.subjects_and_classes[key])] 
+                    if (sched_1.conflicts(sched_2)):
+                        score = -100000
```

Comment:
For good practice, I ask of you to use the negative infinity.
This may be achieved with float("-inf")